src/entities/amm/amm.ts

```diff
@@ -555,47 +303,51 @@ class AMM {
       throw new Error('No underlying error');
     }
 
-    if (validationOnly) {
-      return;
-    }
-
-    const effectiveOwner = !owner ? await this.signer.getAddress() : owner;
-
-    const { closestUsableTick: oldTickUpper } = this.closestTickAndFixedRate(oldFixedLow);
-    const { closestUsableTick: oldTickLower } = this.closestTickAndFixedRate(oldFixedHigh);
+    const owner = await this.signer.getAddress();
 
     const { closestUsableTick: tickUpper } = this.closestTickAndFixedRate(fixedLow);
     const { closestUsableTick: tickLower } = this.closestTickAndFixedRate(fixedHigh);
 
     const factoryContract = factoryFactory.connect(this.factoryAddress, this.signer);
     const peripheryAddress = await factoryContract.periphery();
     const peripheryContract = peripheryFactory.connect(peripheryAddress, this.signer);
-    const _notional = this.scale(notional);
+    const scaledNotional = this.scale(notional);
 
-    let mintOrBurnParams: MintOrBurnParams;
-    let tempOverrides: { value?: BigNumber; gasLimit?: BigNumber } = {};
+    // Get the margin delta in underlying ERC20 tokens and ETH
+    let scaledMarginDelta = '0';
+    const tempOverrides: { value?: BigNumber; gasLimit?: BigNumber } = {};
 
-    if (this.isETH && marginEth) {
-      tempOverrides.value = ethers.utils.parseEther(marginEth.toFixed(18).toString());
-    }
+    if (this.isETH) {
+      // Get the settlement margin from the previous position
 
-    const scaledMarginDelta = this.scale(margin);
+      const settlementMargin = rolloverPosition.settlementBalance;
```

Comment:
small comment, isn't this block of logic the same for swap and rollover, can we abstract it to a separate function to make the code more compact?

Reply:
Yes, that's right.